### PR TITLE
Design picker: Enable categories filter in Gutenboarding

### PR DESCRIPTION
**Pull request by arthur791004**
#### Changes proposed in this Pull Request

* Update the Design Picker in the  /new (aka Gutenboarding) flow to make it consistent with the  Design Picker consistent in the /start flow 
* This PR is for the Gutenboarding flow only 

Prior work on the Design Picker in the /start flow can be found on the [Blogger Experience Pod board](https://github.com/orgs/Automattic/projects/272)

TODO
- [x] Check how to show “Go back” in the design picker of Gutenboarding if enabling the categories filter

**Desktop**

| Before/Disabled | After |
| - | - |
| image | image |

**Mobile**

| Before/Disabled | After |
| - | - |
| image | image |

#### Testing instructions

* Go to `/new/design`
* Check the category filter show on the left side

Related to p1638178895225900/1637819354.184500-slack-C029SB8JT8S

### CODE REVIEW

File: packages/design-picker/src/components/blank-canvas-button/index.tsx
Comment (a8ci18n) [inline packages/design-picker/src/components/blank-canvas-button/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **28** times:
`translate( 'Blank Canvas' )` *ES Score: 8*

File: packages/design-picker/src/components/index.tsx
Comment (p-jackson) [inline packages/design-picker/src/components/index.tsx:20]: We've got to keep exporting this because it's used in the [new page modal](https://github.com/Automattic/wp-calypso/blob/2b61c991f03bbc7ac961f75b183bcd8feba2017d/packages/page-pattern-modal/src/components/pattern-selector-item.tsx#L1) and a [DIFM component](https://github.com/Automattic/wp-calypso/blob/2b61c991f03bbc7ac961f75b183bcd8feba2017d/client/signup/steps/difm-design-picker/demo-tile.tsx#L2).

File: packages/design-picker/src/components/featured-picks-buttons/index.tsx
Comment (p-jackson) [inline packages/design-picker/src/components/featured-picks-buttons/index.tsx]: In the interests of making the design picker the same everywhere, maybe we should just change `blank-canvas`'s title in `available-designs-config.json` to "Blank Canvas".

File: client/landing/gutenboarding/onboarding-block/designs/index.tsx
Comment (p-jackson) [inline client/landing/gutenboarding/onboarding-block/designs/index.tsx]: We may only want to enable the categories when it's _not_ anchor setup. Because I don't think there's very many themes enabled in the anchor flow.

It's been a while since I've tested anchor, but from memory you do it by going to [anchor.fm](anchor.fm) first, creating an account, and then from inside their interface you say you'd like to create a website for your podcast.

File: client/landing/gutenboarding/onboarding-block/designs/index.tsx
Comment (p-jackson) [inline client/landing/gutenboarding/onboarding-block/designs/index.tsx:45]: The options shown in Gutenboarding are slightly different from Themes. I think we want to change Gutenboarding so that it's also grabbing real themes from the API instead of using designs, but in the mean time maybe it's more accurate to keep calling these "designs"?

But maybe the distinction isn't that important because maybe the user doesn't care about the distinction this early in the onboarding.

File: packages/design-picker/src/components/index.tsx
Comment (arthur791004) [reply packages/design-picker/src/components/index.tsx:20]: Actually, it exports from [here](https://github.com/Automattic/wp-calypso/blob/2b61c991f03bbc7ac961f75b183bcd8feba2017d/packages/design-picker/src/index.ts#L3), so we don't need this anymore.

File: packages/design-picker/src/components/featured-picks-buttons/index.tsx
Comment (arthur791004) [reply packages/design-picker/src/components/featured-picks-buttons/index.tsx]: I want to avoid any side effects when we turn off this feature. But it seems we replace the title with “Blank Canvas” everywhere, so maybe we can just rename the title directly 👍

File: client/landing/gutenboarding/onboarding-block/designs/index.tsx
Comment (arthur791004) [reply client/landing/gutenboarding/onboarding-block/designs/index.tsx]: Oh, I've also considered this but I don't notice that we can just disable the category filter here... It's good idea to do it 👍

File: client/landing/gutenboarding/onboarding-block/designs/index.tsx
Comment (arthur791004) [reply client/landing/gutenboarding/onboarding-block/designs/index.tsx:45]: Yep, I think we can just call it “Themes” and make it consistent with Hero Flow.

**Comment by autumnfjeld** [comment]
@arthur791004 I meant to give this a test today, but spent all day in metrics analysis.  Maybe @jordesign could give the /new flow a try to check this Design Picker before you merge.  :)

**Comment by arthur791004** [comment]
@autumnfjeld This doesn't get approval yet, so I don't ping for test.

File: client/landing/gutenboarding/onboarding-block/designs/index.tsx
Comment (p-jackson) [inline client/landing/gutenboarding/onboarding-block/designs/index.tsx]: The categories didn't appear until I made these changes locally.

```suggestion
		! isAnchorFmSignup && isEnabled( 'signup/design-picker-categories' );
```

File: client/landing/gutenboarding/onboarding-block/designs/index.tsx
Comment (p-jackson) [inline client/landing/gutenboarding/onboarding-block/designs/index.tsx]:
```suggestion
		! isAnchorFmSignup && isEnabled( 'signup/design-picker-categories' );
```

**Review by p-jackson — COMMENTED** [review]
Everything's working as expected, except for a little boolean logic error.

I noticed in Ollie's mockups he's generally using two columns of thumbnails, and the width of the category sidebar is wider, like we use in `/start`. Since we're unifying the styles, this could be a good time to roll the style overrides we do in `client/signup/steps/design-picker/style.scss` into the base `@automattic/design-picker` package. So anyone using the picker would get the 2-column of thumbnails by default. I guess we don't need to do this all in this one PR, but it seems like the idea here is to remove some of our unnecessary variations in signup, so pushing our Hero Flow style changes back into the base package could be good.

When we do this we should also use the `highResThumbnails` prop for `<DesignPicker>` since the thumbnails will be larger and we don't want them to be blurry.

I still like having the "Themes" title and subtitle passed in as props (rather than having them baked into the package), because I think it's important for each use of the picker to have control over the copy.

There are also some weird categories i.e. Charity and Non-profit are identical and some categories only have one theme. Perhaps we don't need to fix this in this PR though, and just wait until the picker is API driven instead.

File: client/landing/gutenboarding/onboarding-block/designs/index.tsx
Comment (arthur791004) [reply client/landing/gutenboarding/onboarding-block/designs/index.tsx]: Oops...my brain got stuck at that time 🥲

**Review by p-jackson — APPROVED** [review]
✅ if you want to merge this and maybe we refactor the styles into the package in another PR.

Also happy for you to change styles in this PR too. Ollie's mockups do use a two column layout after all.

**Comment by arthur791004** [comment]
@p-jackson As this PR is a little large, I prefer to refine the styles in another one and create a card to do it: https://github.com/orgs/Automattic/projects/272#card-74252601

**Comment by autumnfjeld** [comment]
@Automattic/ganon-bloggers  I tested the flow and designs and Design Picker categories on /new.  Looks great.  Feel free to 🚢  .

bitmoji

**Pull request merged**
